horace_core/sqw/PixelData/@PixelData/move_to_page.m

```diff
@@ -0,0 +1,39 @@
+function obj = move_to_page(obj, page_number)
+% Set the object to point at the given page number
+%   This function does nothing if the object is not file-backed or is
+%   already on the given page
+%
+page_number = parse_args(obj, page_number);
+
+if obj.is_file_backed_() && obj.page_number_ ~= page_number
+
+    if obj.page_is_dirty_(obj.page_number_) && obj.dirty_page_edited_
+        obj.write_dirty_page_();
+    end
+    obj.page_number_ = page_number;
+    obj.dirty_page_edited_ = false;
+    obj.data_ = zeros(obj.PIXEL_BLOCK_COLS_, 0);
+end
+
+end
+
+
+% -----------------------------------------------------------------------------
+function page_number = parse_args(obj, varargin)
+    parser = inputParser();
+    parser.addRequired('page_number', @is_scalar_positive_int);
+    parser.parse(varargin{:});
+
+    page_number = parser.Results.page_number;
+
+    if page_number > obj.get_num_pages_
+        error('PIXELDATA:move_to_page', ...
+              'Cannot advance to page %i only %i pages of data found.', ...
+              page_number, obj.get_num_pages_);
+    end
+end
+
+
+function is = is_scalar_positive_int(number)
```

Comment:
should the index validators more into a utility folder? we have two in this PR, these may duplicate argument validation we have already implemented and are likely to use again as we move to using `inputParser` for all routines.

while some specialist validation may be required in some functions, the `is_logical_or_vector_of_ints` and `is_positive_integer` are likely to be repeated

Reply:
that would better be done as a TechDebt story after this PR is completed as it requires a review of other inputParsers and changes to code unrelated to PixelData paging

Reply:
My feeling would be the function would be better placed in Herbert anyway - there are some similar utility functions there.

Reply:
https://github.com/pace-neutrons/Horace/issues/414 opened to address this